sdk/servicebus/service-bus/samples/typescript/src/receiveMessagesStreaming.ts

```diff
@@ -30,26 +36,49 @@ export async function main() {
   // - See session.ts for how to receive using sessions.
   const receiver = sbClient.createReceiver(queueName);
 
-  const processMessage = async (brokeredMessage) => {
-    console.log(`Received message: ${brokeredMessage.body}`);
-    await brokeredMessage.complete();
-  };
-  const processError = async (args) => {
-    console.log(`Error from error source ${args.errorSource} occurred: `, args.error);
-  };
-
   try {
-    receiver.subscribe(
+    const subscription = receiver.subscribe(
       {
-        processMessage,
-        processError
+        processMessage: async (brokeredMessage: ServiceBusReceivedMessage) => {
+          console.log(`Received message: ${brokeredMessage.body}`);
+          await receiver.completeMessage(brokeredMessage);
+        },
+        processError: async (args: ProcessErrorArgs) => {
+          console.log(`Error from source ${args.errorSource} occurred: `, args.error);
+
+          // the handler will not stop without explicit intervention from you.
+          if (isMessagingError(args.error)) {
+            switch (args.error.code) {
+              case "MessagingEntityDisabledError":
+              case "MessagingEntityNotFoundError":
+              case "UnauthorizedError":
+                // It's possible you have a temporary infrastructure change (for instance, the entity being
+                // temporarily disabled). The handler will continue to retry - it is completely up to you
+                // what is considered fatal for your program.
+                console.log(
+                  `An unrecoverable error occurred. Stopping processing. ${args.error.code}`,
+                  args.error
+                );
+                await subscription.close();
```

Comment:
What is our recommendation at this point? Close just the subscription and start a new subscribe() call? Close the receiver/client? Or use what ever mechanism you have to denote that nothing can be done here and someone needs to investigate the errors manually?

Reply:
Recommendation should be?
- Not do anything, we'll keep trying to recover everything for you even in the cases where you run into fatal "EntityNotFound" errors... You'll see the receiver perfectly fine once you investigate/fix the errors from your logs
- Close the subscription(or receiver/client), if you want to restart after fixing things on your end

Reply:
I think it really is up to them and there's no reason to speculate. 

The action we have in there now seems sensible enough but if they wanted to they could definitely tear down more.